providers/tezos-provider/src/TezosProvider.ts

```
@@ -0,0 +1,523 @@
+      method: TezosMethod.SEND,
+      params: {
+        account: this.address,
+        operations: [op],
+      },
+    }, this.chainId);
+
+    return result;
+  }
+
+  // Method to send a transaction
+  public async tezosSendTransaction(op: PartialTezosTransactionOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to send a delegation
+  public async tezosSendDelegation(op: PartialTezosDelegationOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to send an undelegation
+  public async tezosSendUndelegation(): Promise<TezosSendResponse> {
+    const op: PartialTezosDelegationOperation = { kind: TezosOperationType.DELEGATION };
+    return this.tezosSend(op);
+  }
+
+  // Method to originate a contract
+  public async tezosSendOrigination(op: PartialTezosOriginationOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to call a smart contract: destination is the contract address, entrypoint as defined in the contract
+  public async tezosSendContractCall(op: PartialTezosTransactionOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Prepare a staking operation: Transaction with destination set to own address, unit as a parameter and specific entrypoint
+  private createStakingOperation(
+    entrypoint: string, 
+    op: PartialTezosTransactionOperation
+  ): PartialTezosTransactionOperation {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return {
+      ...op,
+      kind: TezosOperationType.TRANSACTION,
+      destination: this.address,
+      parameters: {
+        entrypoint,
+        value: { prim: "Unit" },
+      },
+    };
+  }
+
+  public async tezosSendStake(op: PartialTezosTransactionOperation = {
+    kind: TezosOperationType.TRANSACTION,
+    destination: this.address ?? "[initiate provider first]",
+    amount: "1",
```

Amount is not an optional field. The user must provide this while creating the operation.

---

Do not provide default value for amount. Let the transaction fail if the amount is not set in `op`

---

fixed